mteb/tasks/Clustering/eng/StackExchangeClustering.py

```diff
@@ -1,11 +1,75 @@
 from __future__ import annotations
 
+import itertools
+
+from datasets import Dataset, DatasetDict
+
 from mteb.abstasks.TaskMetadata import TaskMetadata
 
 from ....abstasks.AbsTaskClustering import AbsTaskClustering
+from ....abstasks.AbsTaskClusteringFast import AbsTaskClusteringFast
+
+
+class StackExchangeClusteringFast(AbsTaskClusteringFast):
+    metadata = TaskMetadata(
+        name="StackExchangeClusteringFast",
+        description="Clustering of titles from 121 stackexchanges. Clustering of 25 sets, each with 10-50 classes, and each class with 100 - 1000 sentences.",
+        reference="https://arxiv.org/abs/2104.07081",
+        dataset={
+            "path": "mteb/stackexchange-clustering",
+            "revision": "6cbc1f7b2bc0622f2e39d2c77fa502909748c259",
+        },
+        type="Clustering",
+        category="s2s",
+        eval_splits=["test"],
+        eval_langs=["eng-Latn"],
+        main_score="v_measure",
+        date=("2021-01-01", "2021-04-14"),
+        form=["written"],
+        domains=["Web"],
+        task_subtypes=["Thematic clustering"],
+        license="Not specified",
+        socioeconomic_status="mixed",
+        annotations_creators="derived",
+        dialect=[],
+        text_creation="found",
+        bibtex_citation="""@article{geigle:2021:arxiv,
+        author    = {Gregor Geigle and 
+                        Nils Reimers and 
+                        Andreas R{\"u}ckl{\'e} and
+                        Iryna Gurevych},
+        title     = {TWEAC: Transformer with Extendable QA Agent Classifiers},
+        journal   = {arXiv preprint},
+        volume    = {abs/2104.07081},
+        year      = {2021},
+        url       = {http://arxiv.org/abs/2104.07081},
+        archivePrefix = {arXiv},
+        eprint    = {2104.07081}
+        }""",
+        n_samples={"test": 16000},
+        avg_character_length={"test": 57.0},
+    )
+
+    def dataset_transform(self):
+        ds = dict()
+        for split in self.metadata.eval_splits:
+            labels = list(itertools.chain.from_iterable(self.dataset[split]["labels"]))
+            sentences = list(
+                itertools.chain.from_iterable(self.dataset[split]["sentences"])
+            )
+            ds[split] = Dataset.from_dict({"labels": labels, "sentences": sentences})
+        self.dataset = DatasetDict(ds)
+        self.dataset = self.stratified_subsampling(
+            self.dataset,
+            self.seed,
+            self.metadata.eval_splits,
+            label="labels",
+            n_samples=16000,
```

Comment:
Could you please add a run with n_samples=2048 so that we can tell how much speed up we get and the differences in metrics?

Reply:
@isaac-chung Do you want me to generate separate score files for n_samples=2048?

Reply:
Yes please. One of the steps in the guide is
> 2. Subsample if needed
If the dataset is too large, subsample the test set. You are free to select random examples, or use stratified subsampling (should be preferred when possible). The test set should preferably contain no more than 2048 examples.
```
def dataset_transform(self):
    self.dataset = self.stratified_subsampling(
            self.dataset,
            self.seed,
            self.metadata.eval_splits,
            label="labels",
            n_samples=2048,
     )
```

Reply:
Ignore what I said. Going to merge this :)